### DISCUSSION TITLE
Segmenting a long audio file

### DISCUSSION THREAD

**Discussion by dgoryeo** [snakers4/silero-vad#295] [Q&A] [answered]
I have a use case to breal a long audio file into smaller chunks for STT processing. Am I right to think that if I set the threshold parametr to very low (something like 0.2) then VAD would act as a basic segmenting tool in my use case?   Thanks.

**Comment by snakers4** [comment, marked as answer]
You need to plot the probabilities with the in-built tool, and you will be able to eye-ball the probabilities easily.

**Reply by snakers4** [reply]
https://github.com/snakers4/silero-vad/blob/4f57fae3fa68c456758dc64c01ee7ccaf507b5ac/utils_vad.py#L164-L175

**Reply by EtienneAb3d** [reply]
@dgoryeo, I would perhaps rather use a standard threshold, with a high `min_silence_duration_ms`, perhaps 1 or 2 seconds, to cut on long parts without voice. I think it's more pertinent for your STT purpose: you will get more chance to cut between sentences rather than in a middle of them.

**Reply by snakers4** [reply]
It is worth tweaking all of these params for your domain, if you have the time.

**Reply by dgoryeo** [reply]
So far I have tried various permutations of threshold, min_speech, and min_silence. I then compare the results with 2 other results that I have previously generated from split_on_silence and ffmpeg silencedetect. In my test audio, I noticed that there is a spot where any thershold above 0.35 misses --it is a sudden background music behind the dialogue.  

I probably am misusing VAD for my usecase. My intended usecase is to identify and split audio into "scenes" based on their characteristics/content. Then preprocess each scene with suitable parameters for feeding to STT (Whisper in my case). Example if in a scene there is highway noise, the scene gets denoised before feeding to Whisper. Or if quiet talk, it can be preprocessed for loudeness, or if baby crying it can supress that.
